### DISCUSSION TITLE
AssertionError [ERR_ASSERTION]: Cannot wrap non-Error object

### DISCUSSION THREAD

**Discussion by kvbalib** [strapi/strapi#6219] [Legacy Topics]
I'm trying to make the app send an e-mail using built-in provider on POST message type (api/message/controllers/message.js). I altered native create function. It adds the object correctly to the database, the POST request keeps on processing and after a while it throws this error:

```
AssertionError [ERR_ASSERTION]: Cannot wrap non-Error object
      at Function.boomify (node_modules/boom/lib/index.js:107:14)
      at formatBoomPayload (node_modules/strapi/lib/middlewares/boom/index.js:47:22)
      at node_modules/strapi/lib/middlewares/boom/index.js:84:36
      at processTicksAndRejections (internal/process/task_queues.js:97:5)
```

Then the server restarts. I've tried this on production server - it acts the same. I've also rebuilt the app clean, to rule out conflicts factor.

**Expected behavior**
The app was suppose to send an email on POST request with the content of the new item.

**Code snippets**

```
const { parseMultipartData, sanitizeEntity } = require('strapi-utils');

module.exports = {
  async create(ctx) {
    let entity;
    if (ctx.is('multipart')) {
      const { data, files } = parseMultipartData(ctx);
      entity = await strapi.services.message.create(data, { files });
    } else {
      entity = await strapi.services.message.create(ctx.request.body);
    }

    const messageObj = sanitizeEntity(entity, { model: strapi.models.message });

    await strapi.plugins['email'].services.email.send({
      to: 'example@email.com',
      from: messageObj.email,
      replyTo: messageObj.email,
      subject: `Message from ${messageObj.name}`,
      text: messageObj.message,
      html: null,
    });

    return messageObj;
  },
};

```

**System**

- Node.js version: 12.16.3
- NPM version: 6.14.4
- Strapi version: 3.0.0-beta.20.1
- Database: MySQL
- Operating system: OSX 10.15.4

**Comment by derrickmehaffy** [comment]
Hello,

I see you are wanting to ask a question that is not really a bug report or a feature request, questions should be directed to the [discussions](https://github.com/strapi/strapi/discussions)

I have converted this to a discussion for you, we try to keep Github clean and only for bug reports and feature requests.

**Reply by kvbalib** [reply]
I reckon I did everything as official documentation instructs, on a clean Strapi install, so it might be a bug on Strapi's end. Also, I've seen numerous reports about the error, none of them handled :(

Anyway, thanks for the conversion, hope it helps resolving the problem.

**Reply by derrickmehaffy** [reply]
This is usually caused by a failure to send the message (either something is wrong with the mail config, or Strapi couldn't send the mail).

The reason it was moved over to discussions, is it generally is not an issue with Strapi directly but based on user customization. Any level of customization is not generally a "bug" with Strapi (hence discussions). :smiley:

**Comment by lauriejim** [comment]
Can you try catch the function that send email to see the error sent by this function.

**Reply by kvbalib** [reply]
```
error Error: Internal Server Error
    at Object.create (message.js:31:13)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
```

{"messages":[{"id":"Auth.form.error.email.invalid"}]}

**Reply by lauriejim** [reply]
`from: messageObj.message,` are you sure `messageObj.message` is an email?

**Reply by kvbalib** [reply]
@lauriejim You're right, it's not but, that's not the source of the error - I've tried hardcoding strings with email address and it's all the same.

**Reply by thamanchand** [reply]
i am having same issue! Any solution?

**Reply by hkovacs** [reply]
> i am having same issue! Any solution?

please provide your code for troubleshooting.

the issue might be something as simple as ensuring that the `from` and `replyTo` emails match the api service configuration. for sendgrid, if your configured domain is `abc.com` then your emails must match that. dont hard code anyold.com as it just fails...

that was the solution to my AssertionError failure.

**Reply by hkovacs** [reply]
addendum:
i was able to send emails even with custom `replyTo`, so it seems at least for sendgrid, you only need to ensure the `from` email matches the api config.

**Reply by kvbalib** [reply]
I've checked with email address matching Sendmail config and it doesn't fix the problem.

**Reply by hkovacs** [reply]
> I've checked with email address matching Sendmail config and it doesn't fix the problem.

have you tried testing directly with sendgrid?
https://sendgrid.com/docs/for-developers/sending-email/curl-examples/
